### ISSUE TITLE
Using ACS-engine with a preexisting VMSS

### ISSUE THREAD

**Issue by danielepolencic** [Azure/acs-engine#4175]
This is a request for clarification.

I'd like to provision node pools for my Kubernetes cluster before I create it.
Ideally, I'd like to create a few VMSS with Terraform and then use ACS-engine to generate the templates to create a Kubernetes cluster using the newly created agent pools.

As far as I understand, agent pools in a k8s cluster can only be created by ACS-engine. There isn't a way to plug-in existing VMSS.

Use case:
- I'd like to provision agent pool separately
- I'd like to fine-tune the VMSS, add a load balancer, probes, etc.
- I'd like to use the VMSS that I created earlier in my Kubernetes cluster with ACS-engine

I tried to search for similar use cases in the opened and closed issues, but I couldn't find an answer.

**Comment by jackfrancis** [comment]
This isn't possible, and probably isn't something that will happen anytime soon.

Can you clarify the fine-tuning you wish to do? The current architecture of acs-engine sort of suggests that enforcement would arrive in the initial ARM template that acs-engine prepares.

**Comment by danielepolencic** [comment]
> Please note that I use Terraform to provision infrastructure. If I can't script it, I can't deploy to production.

TL;DR: I'd like to configure my infrastructure without ACS-engine getting in the way.

I'd like to configure a VMSS/VMAS in a private vnet and provision a load balancer on top. That's already doable with vanilla Azure. With the VMSS provisioned, I'd like to attach it to a Kubernetes cluster.
I'd like to repeat the previous action for all the node pools that I wish to attach to a cluster.

The advantage of creating my node pools is that I can leverage existing tools and constructs to build a cluster, rather than reinventing the wheel and creating a totally separate way to create new agent pools.

As an example, I can create Autoscaling groups in AWS and attach them to an EKS cluster. I can create a node pool in GCE and attach it to a GKE cluster.
I can create an ALB and attach it to the autoscaling group created previously. Indeed, it's just a regular autoscaling group.

I have all the level of detail that I would normally expect from an autoscaling group, no new API was created. I can leverage the existing tools and knowledge.

In AKS, I can have only 1 agent pool. And I don't have access to it. In ACS-engine I can have multiple node pools, but I have to define the node pools within the cluster. I don't have access to the node pools to provision load balancers.

I understand that you need the node pools defined in advance so that you can install the kubelet. But with the current solution, I'm limited by what ACS-engine can expose (which is not enough).

If I made a mistake, please let me know. I spent some time reading the issues and the code, but I'm no expert.
